administrator/components/com_guidedtours/src/Table/TourTable.php

```
@@ -127,4 +127,32 @@ protected function _getAssetTitle()
     {
         return $this->title;
     }
+
+    /**
+     * Method to get the parent asset under which to register this one.
+     *
+     * By default, all assets are registered to the ROOT node with ID, which will default to 1 if none exists.
+     * The extended class can define a table and id to lookup.  If the asset does not exist it will be created.
+     *
+     * @param   Table    $table  A Table object for the asset parent.
+     * @param   integer  $id     Id to look up
+     *
+     * @return  integer
+     *
+     * @since   4.3.0
+     */
+    // phpcs:ignore
```

??

---

New rules in the drone forbid us to have functions starting with an underscore. However, we use them in the core so that ignore rule needs to apply or the build will fail

---

everytime I see something saying to ignore an error its a red flag to me. either the code is wrong or the rule is wrong and needs to be corrected.

---

@brianteeman That's not anything to be fixed with this PR here. This PR does it in the same way as done elsewhere already.

---

@obuisard Could you remove the `// phpcs:ignore` here at this place and instead add the file to the list of the allowed exceptions here? https://github.com/joomla/joomla-cms/blob/4.3-dev/ruleset.xml#L31 Regarding alpha ordering it should go between line 40 and 41. Then we would be consistent with how we handle that rule's exceptions for other files.

---

Ah, I just we have already other of these exceptions inside the file. If we would want to clean that up we could add it to the ruleset.xml exceptions at the place mentioned in my previous comment and also a bit below here: https://github.com/joomla/joomla-cms/blob/4.3-dev/ruleset.xml#L153 . But I think that would be a bigger clean up, so maybe something for another PR.